### PR TITLE
Feature/restructuring on reset timeout

### PR DESCRIPTION
**Pull request by LitvinenkoIra**
Initial implementation of [https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md)

This PR is **[ready]** for review.

### Risk
This PR makes **[major]** API changes.

### Testing Plan
https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2129

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/include/application_manager/reset_timeout_handler.h [hunk 1]
```diff
@@ -30,41 +30,31 @@
...
+class ResetTimeoutHandler {
+ public:
+  /**
+   * @brief addRequest adds request that waits for response
+   * @param hmi_correlation_id hmi correlation id
+   * @param correlation_id mobile correlation id
+   * @param connection_key app id
+   */
+  virtual void addRequest(uint32_t hmi_correlation_id,
+                          uint32_t correlation_id,
```
Comment (alexkutsan) [inline src/components/include/application_manager/reset_timeout_handler.h]: @LitvinenkoIra mobile_correlation_id

File: src/components/interfaces/HMI_API.xml [hunk 2]
```diff
@@ -4535,7 +4556,7 @@
   <function name="ClosePopUp" messagetype="response">
     <description>Provides the result of operation.</description>
   </function>
-  <function name="OnResetTimeout" messagetype="notification">
+  <function name="OnResetTimeout" messagetype="notification" deprecated="true" since="5.0">
```
Comment (alexkutsan) [inline src/components/interfaces/HMI_API.xml]: @LitvinenkoIra are you sure that it is correct notation? shouldn't it be until='5.0' ?

File: src/components/include/application_manager/reset_timeout_handler.h [hunk 3]
```diff
@@ -30,41 +30,31 @@
...
-}  // namespace hmi
-
-}  // namespace commands
-
+#ifndef SRC_COMPONENTS_INCLUDE_APPLICATION_MANAGER_RESET_TIMEOUT_HANDLER_H_
+#define SRC_COMPONENTS_INCLUDE_APPLICATION_MANAGER_RESET_TIMEOUT_HANDLER_H_
+
+namespace application_manager {
+namespace request_controller {
+class ResetTimeoutHandler {
```
Comment (alexkutsan) [inline src/components/include/application_manager/reset_timeout_handler.h]: @LitvinenkoIra please add description of the class

File: src/components/application_manager/src/reset_timeout_handler_impl.cc [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    requests_.erase(it);
+  };
+}
+
+void ResetTimeoutHandlerImpl::on_event(const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  hmi_apis::FunctionID::eType event_id = event.id();
+  if (hmi_apis::FunctionID::BasicCommunication_OnResetTimeout == event_id) {
+    const smart_objects::SmartObject& message = event.smart_object();
+    LOG4CXX_INFO(logger_, "Received OnResetTimeout event");
```
[the remark this answers is not in the document]
Comment (LitvinenkoIra) [reply src/components/application_manager/src/reset_timeout_handler_impl.cc]: @LuxoftAKutsan _RequestController::updateRequestTimeout_ already contains info log with all necessary information. So I leave log message only in failure case

File: src/components/interfaces/HMI_API.xml (hunk 2 above)
Comment (LitvinenkoIra) [reply src/components/interfaces/HMI_API.xml]: @LuxoftAKutsan I don't see any "until" fields in API.
and these changes I take from original proposal
https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0189-Restructuring-OnResetTimeout.md

File: src/components/include/application_manager/reset_timeout_handler.h (hunk 3 above)
Comment (LitvinenkoIra) [reply src/components/include/application_manager/reset_timeout_handler.h]: @LuxoftAKutsan done. 
653d522

**Comment by LitvinenkoIra** [comment]
@LuxoftAKutsan please take a look at the latest updates.

File: src/components/application_manager/src/reset_timeout_handler_impl.cc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    }
+    auto request_id =
+        message[strings::msg_params][strings::request_id].asUInt();
+    auto it = requests_.find(request_id);
+    if (it != requests_.end()) {
+      auto mob_data = it->second;
+      application_manager_.updateRequestTimeout(
+          mob_data.first, mob_data.second, timeout);
+    }
+    LOG4CXX_INFO(logger_,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/src/reset_timeout_handler_impl.cc]: @LitvinenkoIra does not looks like info log

**Pull request merged**